### DISCUSSION TITLE
Restricting max number of client connection per user.

### DISCUSSION THREAD

**Discussion by divyn10** [pgbouncer/pgbouncer#1079] [Q&A] [answered]
Is it possible to restrict the max number of client connection per user in pgbouncer?

There was an issue observed in one of our setup where one user/application created lot of connection to pgbouncer and not closed those connection hence resulting in using most of the client connection and other applications start showing error because of no connection available.

**Comment by JelteF** [comment, marked as answer]
Yes, you can specify `max_user_connections` in the `[users]` section. https://www.pgbouncer.org/config.html#max_user_connections-1

**Reply by divyn10** [reply]
In the docs it is mentioned that it will restrict the max server connection, but what about the client connection. 
When I set this values, the client connection to pgbouncer on reaching max_user_connection value was waiting for the connection, and if in the mean time lot of connection requests come from the same user (more than max_client_conn) will it start showing no active client connection to other users/app?
